src/assets.rs

```diff
@@ -313,7 +482,31 @@ impl Assets {
         Ok(self.store.revert_version()?)
     }
 
-    /// Create an unverified asset.
+    /// Insert an asset.
+    ///
+    /// If the store doesn't have an asset with the same code, adds the created asset to the store.
+    /// Otherwise, updates the exisiting asset.
+    ///
+    /// Returns the editor for the inserted asset.
+    pub fn insert<L: Ledger>(
+        &mut self,
+        mut asset: Asset,
+    ) -> Result<AssetEditor<'_>, KeystoreError<L>> {
+        // The asset is verified if it's in the verified set.
+        if self.verified_assets.contains(&asset.definition.code) {
+            asset.verified = true
+        }
+        let store_asset = self.store.load(&asset.definition.code);
+        let mut editor = AssetEditor::new(&mut self.store, asset.clone());
+        if store_asset.is_ok() {
+            // Make sure the icon has the default size.
+            editor = editor.update::<L>(asset.clone())?.set_icon(asset.icon());
+        }
+        editor.save::<L>()?;
+        Ok(editor)
+    }
```

Comment:
The "if" and "otherwise" cases are actually both included in the comments, or am I missing something? I should change `created asset` to `inserted asset` though.

Reply:
The comment makes sense, and the name is consistent with how insert into HashMap works so I think I just misread the code